src/modules/job-exec/job-exec.c

```diff
@@ -173,9 +174,9 @@ flux_future_t *jobinfo_shell_rpc_pack (struct jobinfo *job,
     uint32_t rank;
 
     if (asprintf (&shell_topic,
-                  "%ju-shell-%ju.%s",
+                  "%ju-shell-%s.%s",
                   (uintmax_t) job->userid,
-                  (uintmax_t) job->id,
+                  idf58 (job->id),
                   topic) < 0
```

Comment:
intentional or accidental?   Changing topic string?

Reply:
Looks like it is in the right commit, but the commit message doesn't mention that the job-exec module also hand encodes the shell service topic string, so needed an update too. Will fix.

Reply:
Oops sorry,  I think I am completely off base.  I was thinking of the service published in the `guest.exec.eventlog` e.g.
```json
{"timestamp":1686328514.0094779,"name":"shell.init","context":{"service":"5588-shell-193529793164083200","leader-rank":1,"size":1}}
```
The misdirected point I was trying to make was that tools that might have different environments etc. should use what they find in the eventlog so should be immune from this problem.

Reply:
Yeah, all other entities that use the shell service besides the job-exec module currently use the eventlog. We could update the job-exec module, but I think we figured having the job-exec module watch every job's eventlog to wait for the start event, then watch the exec eventlog for the shell service was an unnecessary burden and would impact job throughput.

Reply:
@chu11 does make a good point. However, the job-exec module does directly invoke the job shell through libsubprocess, and uses its current environment as the environment for the job shell. Therefore, I think the likelihood of the job shell running with a different locale or `FLUX_F58_FORCE_ASCII=t` is low.

However, we do allow the job shell to be overridden in jobspec, in that case a user could invoke a job shell that was built in broken locale mode where the ascii `f` is forced. That would be an annoying problem...

Maybe there is something we could do to make it unnecessary for the job-exec module to use the job shell service directly.

Reply:
> yeah, since it's a change of topic string, perhaps should be in one commit together?

Is it not?
```diff
commit 99e433f08bd6bc89d7d380bc8260d37c09c6c9eb
Author: Mark A. Grondona <mark.grondona@gmail.com>
Date:   Mon Jun 12 13:21:44 2023 -0700

    shell: use f58 jobid encoding for shell service
    
    Problem: The shell service name uses the decimal encoding of the
    jobid as the suffix, but using the f58 encoding would make it more
    clear that the service corresponds directly to a job.
    
    Encode the jobid in the shell service name as f58.
    
    Since the job-exec module builds the shell service topic string by
    hand, update it as well.
---
 src/modules/job-exec/job-exec.c | 4 ++--
 src/shell/svc.c                 | 6 ++++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/modules/job-exec/job-exec.c b/src/modules/job-exec/job-exec.c
index c22eb89c3..c363efc73 100644
--- a/src/modules/job-exec/job-exec.c
+++ b/src/modules/job-exec/job-exec.c
@@ -174,9 +174,9 @@ flux_future_t *jobinfo_shell_rpc_pack (struct jobinfo *job,
     uint32_t rank;
 
     if (asprintf (&shell_topic,
-                  "%ju-shell-%ju.%s",
+                  "%ju-shell-%s.%s",
                   (uintmax_t) job->userid,
-                  (uintmax_t) job->id,
+                  idf58 (job->id),
                   topic) < 0
         || ((rank = resource_set_nth_rank (job->R, 0)) == IDSET_INVALID_ID))
         goto out;
diff --git a/src/shell/svc.c b/src/shell/svc.c
index 3f6e4560d..cb0c55d4a 100644
--- a/src/shell/svc.c
+++ b/src/shell/svc.c
@@ -35,6 +35,8 @@
 #include <unistd.h>
 #include <flux/core.h>
 
+#include "src/common/libjob/idf58.h"
+
 #include "internal.h"
 #include "task.h"
 #include "svc.h"
@@ -66,9 +68,9 @@ static int build_topic (struct shell_svc *svc,
 {
     if (snprintf (buf,
                   len,
-                  "%ju-shell-%ju%s%s",
+                  "%ju-shell-%s%s%s",
                   (uintmax_t)svc->uid,
-                  (uintmax_t)svc->shell->info->jobid,
+                  idf58 (svc->shell->info->jobid),
                   method ? "." : "",
                   method ? method : "") >= len) {
         errno = EINVAL;
```

Maybe I'm looking at the wrong thing...

Reply:
> Like admin/user just sets something up differently on one node vs another. Using the decimal jobid would just seem safer.

I _think_ since the job-exec module starts the shell with a subprocess rexec, the job-exec and shell environments should match (except for FLUX_URI or other overrides)?

Reply:
> Is it not?

Hmm I could have swore I saw one of the changes in another commit.. but maybe I mis-saw.  My bad.

Reply:
In any event, it is _probably_ safe to continue with the solution here for now, since it just changes the topic string using the current behavior. Then perhaps we make the change to how the job-exec module gets the shell's topic string in a separate PR, and make sure we do it before the next release. I can open an issue if this is desirable?

Reply:
That sounds OK to me.  (Also sorry for not refreshing the discussion before commenting!)

I wonder if shell rank 0 could just listen for fatal exceptions since job-exec publishes one right before sending the "exception" RPC?

Reply:
I'm not sure why we didn't do it that way at first. I'd have to review how this works, but not all lost shell exceptions should be fatal I thought.

Reply:
sounds good to me